### PR TITLE
Fix "AssertionError: false == true"

### PR DESCRIPTION
**Pull request by facekapow**
This avoids an AssertionError caused by having "http://" (and others) in URL.
Tested to work.

### CODE REVIEW

**Comment by iefserge** [comment]
Check out `url.parse` function, this might be simpler than regex https://nodejs.org/api/url.html#url_url_parse_urlstr_parsequerystring_slashesdenotehost

**Comment by facekapow** [comment]
OK, now using Node's built-in `url.parse()` function to simplify things.

File: lib/connect.js [hunk 1]
```diff
@@ -44,6 +45,12 @@ function openSocket (host, port) {
   if (isIp(host)) {
     inner.open(host, port);
   } else {
+    // if URL contains "http://" or "https://" (etc), remove it.
+    if (host.indexOf('://') != -1) {
+      // http://www.example.com/superexample -> www.example.com
+      // UPDATE: use url.parse (much easier)
+      host = url.parse(host).host;
```
Comment (iefserge) [inline lib/connect.js:51]: `.host` includes custom port like

```
> url.parse('http://localhost:9000').host
'localhost:9000'
```

I think this should be `.hostname`

File: lib/connect.js [hunk 2]
```diff
@@ -44,6 +45,12 @@ function openSocket (host, port) {
   if (isIp(host)) {
     inner.open(host, port);
   } else {
+    // if URL contains "http://" or "https://" (etc), remove it.
+    if (host.indexOf('://') != -1) {
```
Comment (iefserge) [inline lib/connect.js]: This check might not be necessary, because url always needs to contain protocol

File: lib/connect.js (hunk 1 above)
Comment (iefserge) [inline lib/connect.js:51]: `.hostname` could return null if URL is invalid, i guess we could throw an error in this case (or `socket.emit('error', err);`, need to verify what node does in this case)

**Comment by iefserge** [comment]
Oh, I might be wrong, this is `node-net` module and it shouldn't be able to handle http

**Comment by facekapow** [comment]
I'll remove the check. As for HTTP, that's http-node's job.

**Comment by iefserge** [comment]
@ArielAbreu are you sure this function gets url instead of hostname? I did a console.log

``` js
console.log(host);
```

and got `httpbin.org`

**Comment by kesla** [comment]
@arielabreu hey add a test for this (that passes in node/io.js & runtimejs) and this can get merged!

**Comment by iefserge** [comment]
@kesla are you getting URL in connect instead of host?

**Comment by facekapow** [comment]
@iefserge Might be a stupid question, but, did you log it before or after it resets host?

**Comment by kesla** [comment]
Haven't looked at this at all, my point is that I won't take this into consideration before there's a test here.

**Comment by facekapow** [comment]
@iefserge Sorry, getting the same thing. @kesla I have to finish fixing things.

**Comment by facekapow** [comment]
@iefserge How did you get the AssertionError? I did:

``` javascript
http.get('http://httpbin.org/ip', function(res) {
  runtime.tty.print(res.statusCode);
});
```

And got '200' with no AssertionError whatsoever.

**Comment by facekapow** [comment]
I believe this pull request isn't needed. I'll wait if you find out otherwise, but if not, i'll close it.

**Comment by iefserge** [comment]
@ArielAbreu looks like the issue I'm having is unrelated to this, let's close for now. thanks for looking into this anyway :)

**Comment by facekapow** [comment]
You're welcome, I was getting the issue, too. Then realized i'd forgotten to update my runtimejs dependency.

**Pull request closed without merging**